### ISSUE TITLE
inconsistent replies from systemd-resolved

### ISSUE THREAD

**Issue by smurfix** [systemd/systemd#3981]
There appear to be major caching inconsistencies between systemd-resolved and a "real" name server.
### Submission type
- [*] Bug report

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 1.231
### Used distribution

> Raspbian stretch, systemd self-compiled and -installed
### In case of bug report: Expected behaviour you didn't see

> Hostname gets resolved to both IPv4 and IPv6
### In case of bug report: Unexpected behaviour you saw

A few hours ago:

```
# host mirror.de.leaseweb.net
mirror.de.leaseweb.net is an alias for mirror.fra10.de.leaseweb.net.
mirror.fra10.de.leaseweb.net has IPv6 address 2a00:c98:2030:a034::21
# ping6 mirror.de.leaseweb.net
ping6: unknown host mirror.de.leaseweb.net
# ping6 2a00:c98:2030:a034::21
PING 2a00:c98:2030:a034::21 (2a00:c98:2030:a034::21): 56 data bytes
64 bytes from mirror.fra1.de.leaseweb.net: icmp_seq=1 ttl=53 time=13.597 ms
```

Right now, on the affected system, the cname and the IPv6 address is gone but the IPv4 address appeared:

```
# host mirror.de.leaseweb.net
mirror.de.leaseweb.net has address 37.58.58.140
```

However, asking a server with bind9:

```
# host mirror.de.leaseweb.net 10.107.8.1
[…]
mirror.de.leaseweb.net is an alias for mirror.fra10.de.leaseweb.net.
mirror.fra10.de.leaseweb.net has address 37.58.58.140
mirror.fra10.de.leaseweb.net has IPv6 address 2a00:c98:2030:a034::21
```

**Comment by wica128** [comment]
I can confirm smurfix.

OS: Ubuntu 16.04.1 LTS 
systemd --version
systemd 229
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN


On a AWS EC2 which use the AWS DNS services.

minime@ip-10-1-0-109:~$ host mirror.de.leaseweb.net
mirror.de.leaseweb.net is an alias for mirror.fra10.de.leaseweb.net.
mirror.fra10.de.leaseweb.net has address 37.58.58.140
mirror.fra10.de.leaseweb.net has IPv6 address 2a00:c98:2030:a034::21

bb@ip-10-1-0-109:~$ ping host mirror.de.leaseweb.net
ping: unknown host host

And the second time it works.

minime@ip-10-1-0-109:~$ ping mirror.de.leaseweb.net
PING mirror.fra10.de.leaseweb.net (37.58.58.140) 56(84) bytes of data.
64 bytes from mirror.fra10.de.leaseweb.net (37.58.58.140): icmp_seq=1 ttl=52 time=25.7 ms

**Comment by fsateler** [comment]
This can result in the reverse (showing only ipv6) too. 

```

% host deb.debian.org 127.0.0.53
Using domain server:
Name: 127.0.0.53
Address: 127.0.0.53#53
Aliases:

deb.debian.org is an alias for static.debian.org.
static.debian.org has IPv6 address 2001:41c8:1000:21::21:4
static.debian.org has IPv6 address 2001:4f8:8:36::1deb:22
static.debian.org has IPv6 address 2001:610:1908:b000::148:14
static.debian.org mail is handled by 0 .

% host deb.debian.org 8.8.8.8
Using domain server:
Name: 8.8.8.8
Address: 8.8.8.8#53
Aliases:

deb.debian.org is an alias for static.debian.org.
static.debian.org has address 5.153.231.4
static.debian.org has address 140.211.15.34
static.debian.org has address 149.20.20.22
static.debian.org has address 128.31.0.62
static.debian.org has address 130.89.148.14
static.debian.org has IPv6 address 2001:4f8:8:36::1deb:22
static.debian.org has IPv6 address 2001:610:1908:b000::148:14
static.debian.org has IPv6 address 2001:41c8:1000:21::21:4
static.debian.org mail is handled by 0 .
```

The result from the server seems to be different:

```
% dig deb.debian.org @8.8.8.8

; <<>> DiG 9.10.3-P4-Debian <<>> deb.debian.org @8.8.8.8
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 39034
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;deb.debian.org.                        IN      A

;; ANSWER SECTION:
deb.debian.org.         3599    IN      CNAME   static.debian.org.
static.debian.org.      299     IN      A       130.89.148.14
static.debian.org.      299     IN      A       149.20.20.22
static.debian.org.      299     IN      A       140.211.15.34
static.debian.org.      299     IN      A       128.31.0.62
static.debian.org.      299     IN      A       5.153.231.4

;; Query time: 254 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Fri Aug 19 10:07:08 CLST 2016
;; MSG SIZE  rcvd: 144

% dig deb.debian.org @127.0.0.53

; <<>> DiG 9.10.3-P4-Debian <<>> deb.debian.org @127.0.0.53
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 1223
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;deb.debian.org.                        IN      A

;; ANSWER SECTION:
deb.debian.org.         3041    IN      CNAME   static.debian.org.

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Fri Aug 19 10:07:55 CLST 2016
;; MSG SIZE  rcvd: 64

```

**Comment by poettering** [comment]
So, I am pretty sure both of these incidents are just duplicates of #3826 and should be fixed now, by #5276. Closing hence.
